### PR TITLE
[loki-stack] Use grafana.sidecar.datasources.{label,labelValue} for datasources cm metadata

### PR DESCRIPTION
**Pull request by jjaniec**
When using multiple releases of the loki-stack chart, the datasources of the wrong release can be imported into grafana by the sidecar as the grafana_datasource label is the same.

the datasource configmap should instead use the key/value pair defined in the values of the grafana sub-chart if specified: https://github.com/grafana/helm-charts/blob/d7d211d3b8e5d134afcdbbd40d6cd36be73443f2/charts/grafana/values.yaml#L873 and https://github.com/grafana/helm-charts/blob/d7d211d3b8e5d134afcdbbd40d6cd36be73443f2/charts/grafana/values.yaml#L875

Any feedback or suggestions are appreciated :)

### CODE REVIEW

**Comment by jjaniec** [comment]
Any updates ? 👀

File: charts/loki-stack/templates/datasources.yaml [hunk 1]
```diff
@@ -9,7 +9,11 @@ metadata:
     chart: {{ template "loki-stack.chart" . }}
     release: {{ .Release.Name }}
     heritage: {{ .Release.Service }}
+{{- if .Values.grafana.sidecar.datasources.label }}
+    {{ .Values.grafana.sidecar.datasources.label }}: "{{ .Values.grafana.sidecar.datasources.labelValue }}"
+{{- else }}
     grafana_datasource: "1"
+{{- end }}
```
Comment (zanhsieh) [inline charts/loki-stack/templates/datasources.yaml:16]: Please indent and quote.

```suggestion
    {{- if .Values.grafana.sidecar.datasources.label }}
    {{ .Values.grafana.sidecar.datasources.label }}: {{ .Values.grafana.sidecar.datasources.labelValue | quote }}
    {{- else }}
    grafana_datasource: "1"
    {{- end }}
```

**Review by zanhsieh — CHANGES_REQUESTED** [review]

File: charts/loki-stack/templates/datasources.yaml (hunk 1 above)
Comment (jjaniec) [reply charts/loki-stack/templates/datasources.yaml]: It should be ok now

```bash
Validation success! 👍
==> Linting charts/loki-stack

1 chart(s) linted, 0 chart(s) failed
------------------------------------------------------------------------------------------------------------------------
 ✔︎ loki-stack => (version: "2.8.5", path: "charts/loki-stack")
------------------------------------------------------------------------------------------------------------------------
All charts linted successfully
```

Didn't know about chart-testing, looks amazing!
I will use it again for sure so thanks I guess :)

**Review by zanhsieh — APPROVED** [review]

**Comment by eyenx** [comment]
The default `labelValue` in the grafana chart is set to `""` this renders now to `grafana_datasource: ''` 

Is this really what we want? @zanhsieh

**Pull request merged**